src/renderers/webgl/WebGLTextures.js

```diff
@@ -180,16 +180,27 @@ function WebGLTextures( _gl, extensions, state, properties, capabilities, utils,
 
 			extensions.get( 'EXT_color_buffer_float' );
 
-		} else if ( internalFormat === _gl.RGB16F || internalFormat === _gl.RGB32F ) {
-
-			console.warn( 'THREE.WebGLRenderer: Floating point textures with RGB format not supported. Please use RGBA instead.' );
-
 		}
 
 		return internalFormat;
 
 	}
 
+	// Handles WebGL2 RGBFormat fallback - #18858
+
+	function formatFallback ( renderTarget ) {
+
+		if ( renderTarget.texture.format == RGBFormat &&
+			( renderTarget.texture.type == FloatType || renderTarget.texture.type == HalfFloatType ) ) {
```

Comment:
What's the correct code style / indentation for multi-line if-clauses ?

Reply:
I would write everything in one line.

BTW: If the code is used always once, you can inline `formatFallback()` in `setupFrameBufferTexture()`. I

Reply:
Also ensure to use `===` instead of `==`.

Reply:
> I would write everything in one line.

Sure.

> BTW: If the code is used always once, you can inline the code of `formatFallback()` in `setupFrameBufferTexture()`.

It can also be used when for `BufferStorage` that's why I created a function, instead of inlining.

Reply:
> It can also be used for BufferStorage

Can you please explain this use case a bit. I'm afraid I don't understand the relation.

Reply:
I meant that we also had to deal with the same fallback when calling `gl.renderbufferStorage`, because it has the same limitations. But I noticed that I can instead inline the function directly on `setupRenderTarget`.

Reply:
If you clean up the formatting again, the PR should look good.